timemachine/exchange/exchange_mover.py

```diff
@@ -87,108 +95,385 @@ def get_initial_state(water_pdb, mol, ff, seed, nb_cutoff):
-    )
+        integrator = LangevinIntegrator(temperature, dt, 1.0, combined_masses, seed)
 
-    # (YTZ): This is disabled because the initial ligand and starting waters are pre-minimized
-    # print("Minimizing the host system...", end="", flush=True)
-    # host_conf = minimize_host_4d([mol], host_config, ff)
-    # print("Done", flush=True)
+        bond_list = get_bond_list(host_bps[0].potential)
+        group_idxs = get_group_indices(bond_list, len(combined_masses))
 
-    host_conf = solvent_conf
-    combined_conf = np.concatenate([host_conf, get_romol_conf(mol)], axis=0)
+        barostat = MonteCarloBarostat(
+            len(combined_masses), DEFAULT_PRESSURE, temperature, group_idxs, barostat_interval, seed + 1
+        )
 
-    ligand_idxs = np.arange(num_water_atoms, num_water_atoms + mol.GetNumAtoms())
+        # (YTZ): This is disabled because the initial ligand and starting waters are pre-minimized
+        # print("Minimizing the host system...", end="", flush=True)
+        # host_conf = minimize_host_4d([mol], host_config, ff)
+        # print("Done", flush=True)
 
-    initial_state = InitialState(
-        host_bps, integrator, barostat, combined_conf, np.zeros_like(combined_conf), solvent_box, 0.0, ligand_idxs
-    )
+        host_conf = solvent_conf
+        combined_conf = np.concatenate([host_conf, get_romol_conf(mol)], axis=0)
+
+        ligand_idxs = np.arange(num_water_atoms, num_water_atoms + mol.GetNumAtoms())
+
+        initial_state = InitialState(
+            host_bps, integrator, barostat, combined_conf, np.zeros_like(combined_conf), solvent_box, 0.0, ligand_idxs
+        )
+
+    else:
+        host_bps, host_masses = openmm_deserializer.deserialize_system(solvent_sys, cutoff=nb_cutoff)
+        integrator = LangevinIntegrator(temperature, dt, 1.0, host_masses, seed)
+        bond_list = get_bond_list(host_bps[0].potential)
+        group_idxs = get_group_indices(bond_list, len(host_masses))
+        barostat_interval = 25
+        barostat = MonteCarloBarostat(
+            len(host_masses), DEFAULT_PRESSURE, temperature, group_idxs, barostat_interval, seed + 1
+        )
+
+        ligand_idxs = [0, 1, 2]  # pick first water molecule to be a "ligand" for targetting purposes
+        initial_state = InitialState(
+            host_bps, integrator, barostat, solvent_conf, np.zeros_like(solvent_conf), solvent_box, 0.0, ligand_idxs
+        )
 
     num_water_mols = num_water_atoms // 3
 
     return initial_state, num_water_mols, solvent_topology
 
 
-class ExchangeMove(moves.MonteCarloMove):
-    def __init__(self, nb_params, nb_beta, nb_cutoff, water_idxs):
+from scipy.special import logsumexp
+
+
+class BDExchangeMove(moves.MonteCarloMove):
+    def __init__(
+        self,
+        nb_beta: float,
+        nb_cutoff: float,
+        nb_params: NDArray,
+        water_idxs: NDArray,
+        beta: float,
+    ):
```

Comment:
question: would it make sense to parameterize the **amount of biasing**?

i.e. expose a constructor parameter that when set to 0.0 gives "unbiased" deletion, 1.0 gives the current behavior, and other real values give different behavior. might be interesting to see a curve of MC performance vs. this parameter, and see if it's roughly maximized at 1.0.

Reply:
That's a good question - I haven't investigated this yet. Mainly for simplicity, since keeping it at 1.0 gives you the exact cancellation with the Metropolis component. FWIW, most of the time, the same one or two waters are selected (since the water energies follow a gaussian-like distribution), so the high energy tail is typically only occupied by a very few number of waters.